rest-fights/src/main/java/io/quarkus/sample/superheroes/fight/client/VillainClient.java

```diff
@@ -65,9 +65,12 @@ public VillainClient(FightConfig fightConfig) {
    * Calls hello on the Villains service.
    * @return A "hello" from Villains
    */
-  public String helloVillains() {
+  public CompletionStage<String> helloVillains() {
```

Comment:
This doesn't need to be a `CompletionStage`. A `Uni` is fine.

Reply:
Hum... Well, if I change it to a Uni it doesn't work. This is what I have:

```
  public Uni<String> helloVillains() {
    return this.villainClient.path("hello")
      .request(MediaType.TEXT_PLAIN_TYPE)
      .rx(UniInvoker.class)
      .get(String.class)
      .onFailure(Is404Exception.IS_404).recoverWithNull()
      .subscribeAsCompletionStage();
  }
  ```